src/Status/Api.elm

```diff
@@ -0,0 +1,26 @@
+module Status.Api exposing (getStatus)
+
+import Utils.Api exposing (baseUrl)
+import Http
+import Status.Types exposing (StatusMsg(NewStatus), StatusResponse)
+import Json.Decode exposing (Decoder, map2, string, field, at)
+import Types exposing (Msg(MsgForStatus))
+
+
+getStatus : Cmd Msg
+getStatus =
+    let
+        url =
+            String.join "/" [ baseUrl, "status" ]
+
+        request =
+            Http.get url decodeStatusResponse
```

Comment:
This is open to debate, but the pattern I'm using for handling http requests is described here: http://blog.jenkster.com/2016/06/how-elm-slays-a-ui-antipattern.html

and implemented here: https://github.com/stuartnelson3/am-ui/blob/master/src/Utils/Api.elm#L10-L22

See it in use here: https://github.com/stuartnelson3/am-ui/blob/master/src/Silences/Api.elm#L18-L19

The `Maybe` you're using accomplishes essentially the same thing, but using `ApiData` we have 3 states (and the option for more) so that we can differentiate between success (`Just` in a maybe), failure (`Nothing` in a maybe), and loading (not represented). The app then just relies on this union type to delay all the decision making (what is my actual data type) until it needs to happen, and it also has context into the specific failure reason (which can then be displayed to the user https://github.com/stuartnelson3/am-ui/blob/master/src/Utils/Views.elm#L83-L106).

In `update` we're storing the API response as `ApiData Silence`, and we don't actually worry about what we have until we render a view, or create a silence, for example.

It is a bit verbose (https://github.com/stuartnelson3/am-ui/blob/master/src/Silences/Views.elm), but I'm liking how it encapsulates the different states the different objects can be in, and makes it easy to manage potential failures across multiple different objects if we make multiple API requests.